singer_sdk/helpers/_typing.py

```diff
@@ -542,9 +543,32 @@ def _conform_primitive_property(  # noqa: PLR0911
         # for BIT value, treat 0 as False and anything else as True
         return elem != b"\x00" if is_boolean_type(property_schema) else elem.hex()
     if isinstance(elem, (float, decimal.Decimal)):
-        if math.isnan(elem) or math.isinf(elem):
-            return None
-        return elem
+        return elem if math.isfinite(elem) else None
+    if isinstance(elem, str) and not is_string_type(property_schema):
```

Comment:
(commenting here to start a thread)

> It looks like it was slightly affected: https://codspeed.io/meltano/sdk/branches/ReubenFrankel%3Afeat%2Fconform-primitive-interpret-string?uri=tests%2Fcore%2Ftest_typing.py%3A%3Atest_bench_conform_record_data_types
> 
> Possibly due to this new check, which will check the property schema type for every `str` value:
> 
> https://github.com/meltano/sdk/blob/6c0f4748df1e5e094fdc8f2cdc60477df9c2d65a/singer_sdk/helpers/_typing.py#L546

Do you think there's a way we could prevent this regression? Perhaps caching a mapping of property > is_<type>.

Reply:
Yes, I was thinking about some kind of type caching also. We did something like that [here](https://github.com/Matatika/tap-callminer/blob/e6d59a22d586e16ad01f859dc0043de5b3eef396/tap_callminer/streams.py#L217-L226), or were you imagining it to be per-property?

Reply:
Yeah, something like or like the [selection mask](https://github.com/meltano/sdk/blob/c585ac6df164c83eb19f5b6b24fa6a1d6a5e6398/singer_sdk/singerlib/catalog.py#L21) we use internally here.